lib/her/model/attributes.rb

```diff
@@ -69,48 +79,55 @@ def respond_to_missing?(method, include_private = false)
       #   user.assign_attributes(name: "Lindsay")
       #   user.changes # => { :name => ["Tobias", "Lindsay"] }
       def assign_attributes(new_attributes)
-        @attributes ||= attributes
+        @_attributes ||= attributes
         # Use setter methods first
         unset_attributes = self.class.use_setter_methods(self, new_attributes)
 
         # Then translate attributes of associations into association instances
         parsed_attributes = self.class.parse_associations(unset_attributes)
 
-        # Then merge the parsed_data into @attributes.
-        @attributes.merge!(parsed_attributes)
+        # Then merge the parsed_data into @_attributes.
+        @_attributes.merge!(parsed_attributes)
       end
       alias attributes= assign_attributes
 
       def attributes
-        @attributes ||= HashWithIndifferentAccess.new
+        # The natural choice of instance variable naming here would be
+        # `@attributes`. Unfortunately that causes a naming clash when
+        # used with `ActiveModel` version >= 5.2.0.
+        # As of v5.2.0 `ActiveModel` checks to see if `ActiveRecord`
+        # attributes exist, assuming that if the instance variable
+        # `@attributes` exists on the instance, it is because they are
+        # `ActiveRecord` attributes.
+        @_attributes ||= HashWithIndifferentAccess.new
```

Comment:
Makes sense. We have other examples where instance variables follow a `@_<variable>` format but they're prefixed with `@_her_<variable>`. Not a strong justification but following that format would give us consistency.